File: rad_anastasiya/bridge/src/com/universiry/bridge/shape/Triangle.java
package com.universiry.bridge.shape;

import com.universiry.bridge.color.IColor;

public class Triangle extends Shape {
    private final double first;
    private final double second;
    private final double third;

    public Triangle(final double first, final double second, final double third, final IColor IColor) {
        super(IColor);
        this.first = first;
        this.second = second;
        this.third = third;
    }


    @Override
    public double square() {
        return Math.sqrt(perimeter() * (perimeter() - first) * (perimeter() - second) * (perimeter() - third));
    }

    @Override
    public double perimeter() {
        return first + second + third;
    }

    @Override
    public String toString() {
        return "Triangle{" +
                "first: " + first +
                ", second: " + second +
                ", third: " + third +
                ", color: " + IColor.name()
                + ", perimeter: " + perimeter()
                + ", square: " + square()
                + '}';
    }
}

File: kulikovskiy_georgiy/Bridge/Shape/Triangle.cs
namespace Bridge
{
    public class Triangle : Shape
    {
        public double SideTwo { get; set; }
        public double SideThree { get; set; }

        public Triangle(double sideOne, double sideTwo, double sideThree, IColor color)
            : base(sideOne, color)
        {
            SideTwo = sideTwo;
            SideThree = sideThree;
        }

        public override double Perimeter()
        {
            return SideOne + SideTwo + SideThree;
        }

        public override double Area()
        {
            double semiPerimeter = Perimeter() / 2;
            return Math.Sqrt(
                semiPerimeter
                    * (semiPerimeter - SideOne)
                    * (semiPerimeter - SideTwo)
                    * (semiPerimeter - SideThree)
            );
        }

        public override string ToString()
        {
            return "Triangle: "
                + $"side one: {SideOne}, "
                + $"side two: {SideTwo}, "
                + $"side three: {SideThree}, "
                + $"perimeter: {Perimeter()}, "
                + $"area: {Area()}, "
                + $"color: {Color.GetColor()}";
        }
    }
}
